### PR TITLE
[Refactoring] Remove Braket backend

### PR DESCRIPTION
**Pull request by chMoussa**

### CODE REVIEW

File: tests/qadence/test_overlap.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def _generate_parameter_values() -> tuple[dict, dict]:
     return values_bra, values_ket
 
 
-@pytest.mark.parametrize("backend_name", [BackendName.PYQTORCH, BackendName.BRAKET])
+@pytest.mark.parametrize("backend_name", [BackendName.PYQTORCH])
```
Comment (RolandMacDoland) [inline tests/qadence/test_overlap.py:55]: Is it still necessary to parametrise for a single backend ?

File: tests/qadence/test_overlap.py [hunk 2]
```diff
@@ -125,15 +125,15 @@ def test_overlap_comp_uncomp_exact(backend_name: BackendName, n_qubits: int) ->
...
+        diff_mode=DiffMode.AD,
     )
     ovrlp_cu = ovrlp(values_bra, values_ket)
 
     assert torch.all(torch.isclose(ovrlp_theor, ovrlp_cu, atol=LOW_ACCEPTANCE))
 
 
 @pytest.mark.flaky(max_runs=5)
-@pytest.mark.parametrize("backend_name", [BackendName.PYQTORCH, BackendName.BRAKET])
+@pytest.mark.parametrize("backend_name", [BackendName.PYQTORCH])
```
Comment (RolandMacDoland) [inline tests/qadence/test_overlap.py:136]: Same.

**Review by RolandMacDoland — COMMENTED** [review]
Thanks @chMoussa it feels so much cleaner. Minor comments about parametrisation for a single backend. Will approve soon after.

**Comment by chMoussa** [comment]
> Thanks @chMoussa it feels so much cleaner. Minor comments about parametrisation for a single backend. Will approve soon after.

No it was not. I just removed every parameterization with a single backend and if in a near future we start adding more backends, we can go back to this MR and reput them back.

**Review by RolandMacDoland — APPROVED** [review]
Nice work, thanks @chMoussa.

**Pull request merged**